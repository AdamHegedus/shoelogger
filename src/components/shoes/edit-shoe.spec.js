import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import BootstrapVue from 'bootstrap-vue';
import EditShoe from '@/components/shoes/edit-shoe';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(BootstrapVue);

describe('edit-shoe.vue', () => {
    describe('meta.status is false', () => {
        const $route = {
            params: {
                id: '1'
            }
        };

        const shoesGetters = {
            getMeta: () => {
                return {
                    status: false,
                    message: 'Error message text.'
                };
            }
        };

        const shoesActions = {
            editShoe: jest.fn(),
            getShoesById: jest.fn(() => {
                return {
                    id: 1,
                    typeId: 0,
                    brandId: 0,
                    product: 'Foo Bar'
                };
            })
        };

        const brandsActions = {
            getBrands: jest.fn()
        };

        const typesActions = {
            getTypes: jest.fn()
        };

        const typesState = {
            types: [
                {
                    id: 0,
                    type: 'Foo'
                }
            ]
        };

        const brandsState = {
            brands: [
                {
                    id: 0,
                    brand: 'Foo'
                }
            ]
        };

        const store = new Vuex.Store({
            modules: {
                shoes: {
                    namespaced: true,
                    getters: shoesGetters,
                    actions: shoesActions
                },
                brands: {
                    namespaced: true,
                    state: brandsState,
                    actions: brandsActions
                },
                types: {
                    namespaced: true,
                    state: typesState,
                    actions: typesActions
                }
            }
        });

        it('should match snapshot when page loaded', () => {
            // ASSIGN
            const wrapper = shallowMount(EditShoe, {
                store,
                localVue,
                mocks: {
                    $route
                }
            });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('meta.status is true', () => {
        const $route = {
            params: {
                id: '1'
            }
        };

        const shoesGetters = {
            getMeta: () => {
                return {
                    status: true,
                    message: null
                };
            }
        };

        const shoesActions = {
            editShoe: jest.fn(),
            getShoesById: jest.fn(() => {
                return {
                    id: 1,
                    typeId: 0,
                    brandId: 0,
                    product: 'Foo Bar'
                };
            })
        };

        const brandsActions = {
            getBrands: jest.fn()
        };

        const typesActions = {
            getTypes: jest.fn()
        };

        const typesState = {
            types: [
                {
                    id: 0,
                    type: 'Foo'
                }
            ]
        };

        const brandsState = {
            brands: [
                {
                    id: 0,
                    brand: 'Foo'
                }
            ]
        };

        const store = new Vuex.Store({
            modules: {
                shoes: {
                    namespaced: true,
                    getters: shoesGetters,
                    actions: shoesActions
                },
                brands: {
                    namespaced: true,
                    state: brandsState,
                    actions: brandsActions
                },
                types: {
                    namespaced: true,
                    state: typesState,
                    actions: typesActions
                }
            }
        });

        it('should match snapshot when page loaded', () => {
            // ASSIGN
            const wrapper = shallowMount(EditShoe, {
                store,
                localVue,
                mocks: {
                    $route
                }
            });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    it('should match snapshot when text input changed', () => {
        // ASSIGN
        const $route = {
            params: {
                id: '1'
            }
        };

        const shoesGetters = {
            getMeta: () => {
                return {
                    status: true,
                    message: null
                };
            }
        };

        const shoesActions = {
            editShoe: jest.fn(),
            getShoesById: jest.fn(() => {
                return {
                    id: 1,
                    typeId: 0,
                    brandId: 0,
                    product: 'Foo Bar'
                };
            })
        };

        const brandsActions = {
            getBrands: jest.fn()
        };

        const typesActions = {
            getTypes: jest.fn()
        };

        const typesState = {
            types: [
                {
                    id: 0,
                    type: 'Foo'
                }
            ]
        };

        const brandsState = {
            brands: [
                {
                    id: 0,
                    brand: 'Foo'
                }
            ]
        };

        const store = new Vuex.Store({
            modules: {
                shoes: {
                    namespaced: true,
                    getters: shoesGetters,
                    actions: shoesActions
                },
                brands: {
                    namespaced: true,
                    state: brandsState,
                    actions: brandsActions
                },
                types: {
                    namespaced: true,
                    state: typesState,
                    actions: typesActions
                }
            }
        });

        const wrapper = shallowMount(EditShoe, {
            store,
            localVue,
            mocks: {
                $route
            }
        });

        const expected = 'Test';
        const input = 'Test';

        // ACT
        wrapper.find('input').setValue(input);

        // ASSERT
        const element = wrapper.element;
        const actual = wrapper.vm.$data.product;
        expect(element).toMatchSnapshot();
        expect(actual).toEqual(expected);
    });

    describe('getShoe', () => {
        it('should return the value of text input', () => {
            // ASSIGN
            const $route = {
                params: {
                    id: '1'
                }
            };

            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                editShoe: jest.fn(),
                getShoesById: jest.fn(() => {
                    return {
                        id: 1,
                        typeId: 0,
                        brandId: 0,
                        product: 'Foo Bar'
                    };
                })
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 0,
                        type: 'Foo'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const shoesState = {
                shoe: {
                    id: 1,
                    typeId: 0,
                    brandId: 0,
                    product: 'Foo Bar'
                }
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions,
                        state: shoesState
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(EditShoe, {
                store,
                localVue,
                mocks: {
                    $route
                }
            });

            const input = 'Test';
            wrapper.vm.$data.selectedType = {
                id: 0,
                type: 'Foo'
            };
            wrapper.vm.$data.selectedBrand = {
                id: 0,
                type: 'Foo'
            };
            wrapper.find('#productName').setValue(input);

            // ACT
            const actual = wrapper.vm.getShoe;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('brands', () => {
        it('should return the value from store', () => {
            // ASSIGN
            const $route = {
                params: {
                    id: '1'
                }
            };

            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                editShoe: jest.fn(),
                getShoesById: jest.fn(() => {
                    return {
                        id: 1,
                        typeId: 0,
                        brandId: 0,
                        product: 'Foo Bar'
                    };
                })
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 0,
                        type: 'Foo'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(EditShoe, {
                store,
                localVue,
                mocks: {
                    $route
                }
            });

            // ACT
            const actual = wrapper.vm.brands;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('types', () => {
        it('should return the value from store', () => {
            // ASSIGN
            const $route = {
                params: {
                    id: '1'
                }
            };

            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                editShoe: jest.fn(),
                getShoesById: jest.fn(() => {
                    return {
                        id: 1,
                        typeId: 0,
                        brandId: 0,
                        product: 'Foo Bar'
                    };
                })
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 0,
                        type: 'Foo'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(EditShoe, {
                store,
                localVue,
                mocks: {
                    $route
                }
            });

            // ACT
            const actual = wrapper.vm.types;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('editShoe', () => {
        it('should call action', () => {
            // ASSIGN
            const $route = {
                params: {
                    id: '1'
                }
            };

            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                editShoe: jest.fn(),
                getShoesById: jest.fn(() => {
                    return {
                        id: 1,
                        typeId: 0,
                        brandId: 0,
                        product: 'Foo Bar'
                    };
                })
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 1,
                        type: 'Bar'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(EditShoe, {
                store,
                localVue,
                mocks: {
                    $route
                }
            });
            const input = {
                id: 1,
                brandId: 0,
                typeId: 1,
                product: 'Test'
            };

            // ACT
            wrapper.vm.editShoe(input);

            // ASSERT
            const actual = shoesActions.editShoe.mock.calls[
                shoesActions.editShoe.mock.calls.length - 1
            ][1];
            expect(actual).toMatchSnapshot();
        });
    });

    describe('selectBrand', () => {
        it('should set selectedBrand', () => {
            // ASSIGN
            const $route = {
                params: {
                    id: '1'
                }
            };

            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                editShoe: jest.fn(),
                getShoesById: jest.fn(() => {
                    return {
                        id: 1,
                        typeId: 0,
                        brandId: 0,
                        product: 'Foo Bar'
                    };
                })
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 1,
                        type: 'Bar'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(EditShoe, {
                store,
                localVue,
                mocks: {
                    $route
                }
            });
            const input = {
                id: 0,
                brand: 'Foo'
            };

            // ACT
            wrapper.vm.selectBrand(input);

            // ASSERT
            const actual = wrapper.vm.selectedBrand;
            expect(actual).toMatchSnapshot();
        });
    });

    describe('selectType', () => {
        it('should set selectedType', () => {
            // ASSIGN
            const $route = {
                params: {
                    id: '1'
                }
            };

            const shoesGetters = {
                getMeta: () => {
                    return {
                        status: true,
                        message: null
                    };
                }
            };

            const shoesActions = {
                editShoe: jest.fn(),
                getShoesById: jest.fn(() => {
                    return {
                        id: 1,
                        typeId: 0,
                        brandId: 0,
                        product: 'Foo Bar'
                    };
                })
            };

            const brandsActions = {
                getBrands: jest.fn()
            };

            const typesActions = {
                getTypes: jest.fn()
            };

            const typesState = {
                types: [
                    {
                        id: 1,
                        type: 'Bar'
                    }
                ]
            };

            const brandsState = {
                brands: [
                    {
                        id: 0,
                        brand: 'Foo'
                    }
                ]
            };

            const store = new Vuex.Store({
                modules: {
                    shoes: {
                        namespaced: true,
                        getters: shoesGetters,
                        actions: shoesActions
                    },
                    brands: {
                        namespaced: true,
                        state: brandsState,
                        actions: brandsActions
                    },
                    types: {
                        namespaced: true,
                        state: typesState,
                        actions: typesActions
                    }
                }
            });

            const wrapper = shallowMount(EditShoe, {
                store,
                localVue,
                mocks: {
                    $route
                }
            });
            const input = {
                id: 1,
                brand: 'Bar'
            };

            // ACT
            wrapper.vm.selectType(input);

            // ASSERT
            const actual = wrapper.vm.selectedType;
            expect(actual).toMatchSnapshot();
        });
    });
});
