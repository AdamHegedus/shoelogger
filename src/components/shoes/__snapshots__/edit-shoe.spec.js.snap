// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit-shoe.vue brands should return the value from store 1`] = `
Array [
  Object {
    "brand": "Foo",
    "id": 0,
  },
]
`;

exports[`edit-shoe.vue editShoe should call action 1`] = `
Object {
  "brandId": 0,
  "id": 1,
  "product": "Test",
  "typeId": 1,
}
`;

exports[`edit-shoe.vue getBrandDisplayValue should return correct value 1`] = `"Bar"`;

exports[`edit-shoe.vue getShoe should return the value of text input 1`] = `
Object {
  "brandId": 0,
  "id": 1,
  "product": "Test",
  "typeId": 0,
}
`;

exports[`edit-shoe.vue getTypeDisplayValue should return correct value 1`] = `"Bar"`;

exports[`edit-shoe.vue meta.status is false should match snapshot when page loaded 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Edit Shoe
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label>
            Type
          </label>
           
          <dropdown-stub
            data="[object Object]"
            event="select-type"
            getdisplayvalue="function () { [native code] }"
            placeholder="Select Type"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label>
            Brand Name
          </label>
           
          <dropdown-stub
            data="[object Object]"
            event="select-brand"
            getdisplayvalue="function () { [native code] }"
            placeholder="Select Brand"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="productName"
          >
            Product Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="productName"
            placeholder="Product Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          disabled="disabled"
          id="save"
          type="submit"
        >
          
                    Save
                
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`edit-shoe.vue meta.status is true should match snapshot when page loaded 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Edit Shoe
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label>
            Type
          </label>
           
          <dropdown-stub
            data="[object Object]"
            event="select-type"
            getdisplayvalue="function () { [native code] }"
            placeholder="Select Type"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label>
            Brand Name
          </label>
           
          <dropdown-stub
            data="[object Object]"
            event="select-brand"
            getdisplayvalue="function () { [native code] }"
            placeholder="Select Brand"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="productName"
          >
            Product Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="productName"
            placeholder="Product Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          disabled="disabled"
          id="save"
          type="submit"
        >
          
                    Save
                
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`edit-shoe.vue selectBrand should set selectedBrand 1`] = `
Object {
  "brand": "Foo",
  "id": 0,
}
`;

exports[`edit-shoe.vue selectType should set selectedType 1`] = `
Object {
  "id": 1,
  "type": "Bar",
}
`;

exports[`edit-shoe.vue should match snapshot when text input changed 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h2>
        Edit Shoe
      </h2>
       
      <form>
        <div
          class="form-group"
        >
          <label>
            Type
          </label>
           
          <dropdown-stub
            data="[object Object]"
            event="select-type"
            getdisplayvalue="function () { [native code] }"
            placeholder="Select Type"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label>
            Brand Name
          </label>
           
          <dropdown-stub
            data="[object Object]"
            event="select-brand"
            getdisplayvalue="function () { [native code] }"
            placeholder="Select Brand"
          />
        </div>
         
        <div
          class="form-group"
        >
          <label
            for="productName"
          >
            Product Name
          </label>
           
          <input
            class="form-control form-control-lg shadow border border-primary"
            id="productName"
            placeholder="Product Name"
            type="text"
          />
        </div>
         
        <button
          class="btn btn-primary btn-lg btn-block shadow"
          disabled="disabled"
          id="save"
          type="submit"
        >
          
                    Save
                
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`edit-shoe.vue types should return the value from store 1`] = `
Array [
  Object {
    "id": 0,
    "type": "Foo",
  },
]
`;
